apps/pipelines/nodes/base.py

```diff
@@ -18,10 +18,17 @@ def add_messages(left: list, right: list):
     return left + right
 
 
-class PipelineState(TypedDict):
+class PipelineState(dict):
     messages: Annotated[Sequence[Any], add_messages]
     experiment_session: ExperimentSession
```

Comment:
Putting these on a normal dict doesn't work. I think you should convert this to a pydantic model: https://langchain-ai.github.io/langgraph/how-tos/state-model/#define-the-agent-state

Reply:
> Putting these on a normal dict doesn't work.

Not sure I follow? It does actually work. I am able to run pipelines with this

Reply:
Interesting, I wonder if that will change when we upgrade langgraph.

What I mean by 'does't work' is that TypedDict is a special class that just uses those fields for type validation. I'm surprised that putting them on a normal dict class has any effect.

Reply:
To be honest, I was also surprised

Reply:
I can change this as part of my followup PR if it makes you nervous. I must admit, now that you mentioned this it sort of makes me nervous